libraries/AP_BattMonitor/AP_BattMonitor_MAVLink_Battery.cpp

```diff
@@ -0,0 +1,129 @@
+#include <AP_HAL/AP_HAL.h>
+#include <GCS_MAVLink/GCS_MAVLink.h>
+#include "AP_BattMonitor_MAVLink_Battery.h"
+
+AP_BattMonitor_MAVLink_Battery::AP_BattMonitor_MAVLink_Battery(AP_BattMonitor &mon,
+                                             AP_BattMonitor::BattMonitor_State &mon_state,
+                                             AP_BattMonitor_Params &params) :
+    AP_BattMonitor_Backend(mon, mon_state, params)
+{
+	_state.healthy = false;
+}
+
+void AP_BattMonitor_MAVLink_Battery::read()
+{
+    if ((AP_HAL::micros() - _state.last_time_micros) > MAVLINK_SMART_BATTERY_TIMEOUT_MICROS) {
+		_state.healthy = false;
+		return;
+	} 
+	_state.healthy = true;	
+   return;
+}
+
+bool AP_BattMonitor_MAVLink_Battery::capacity_remaining_pct(uint8_t &cap_remain_pct) const
+{
+    if (capacity_rem_pct != INT8_MIN) {
+		/* positive capacity remaining is expected */
+		if (capacity_rem_pct > 0) {
+			cap_remain_pct = capacity_rem_pct;
+		} else {
+			cap_remain_pct = 0;
+		}
+    	return true;
+	}
+	return false;
+}
+
+bool AP_BattMonitor_MAVLink_Battery::get_cycle_count(uint16_t &cycles) const
+{
+    if (cycle_count != UINT16_MAX) {
+		cycles = cycle_count;
+    	return true;
+	}
+	return false;
+}
+
+void AP_BattMonitor_MAVLink_Battery::process_info_msg(const mavlink_message_t &msg)

```

Comment:
This packet gives us the ability to see the string from the vendor via device_name. Would be nice to do a send-once type of gcs>send_text() of these strings

Reply:
in my case i support standard batteries 3 to 10 cells - not supporting a custom device_name because i can't detect a switched battery; but it would be a nice feature for my product to implement a battery database which loads different parameters and history for a certain device_name/serial number; right now it's intended use is to prevent the need for removing the batteries - 2 wire charging both batteries with an ordinary lixx charger (cc - cv source); 
you can set the device into sleep mode (~1mA) - so theoretically you would never have to remove the batteries;
 
generally, smart_bat_info_msg is being forwarded to the GCS right now, so the device name would already be at the GCS;